### PR TITLE
ISPN-8396 Add interceptor preventing going out of memory

### PR DESCRIPTION
**Pull request by wburns**
* Add strategy type configuration
* Add documentation describing strategies

https://issues.jboss.org/browse/ISPN-8396

### CODE REVIEW

File: core/src/main/java/org/infinispan/util/logging/Log.java [hunk 1]
```diff
@@ -1748,4 +1749,10 @@ CacheConfigurationException offHeapMemoryEvictionSizeNotLargeEnoughForAddresses(
 
    @Message(value = "Cannot acquire lock '%s' for persistent global state", id = 512)
    CacheConfigurationException globalStateCannotAcquireLockFile(@Cause Throwable cause, File lockFile);
+
+   @Message(value = "Memory based eviction requires a transactional cache that doesn't allow for 1 phase commit or synchronizations", id = 513)
```
Comment (gustavonalle) [inline core/src/main/java/org/infinispan/util/logging/Log.java]: Should it be "Exception based eviction requires (...)" ?

Comment (wburns) [reply core/src/main/java/org/infinispan/util/logging/Log.java]: Yes, thanks.

**Comment by wburns** [comment]
Updated and rebased

**Comment by gustavonalle** [comment]
Merged, thanks @wburns !

**Pull request merged**